src/App.tsx

```diff
@@ -15,48 +16,117 @@ export const goodsFromServer = [
+
+  function getReorderedGoods(goods: string[]) {
+    const visibleGoods = [...goods];
+
+    switch (sortType) {
+      case SortType.ALPHABET:
+        visibleGoods.sort((firstGood, secondGood) => (
+          firstGood.localeCompare(secondGood)
+        ));
+        break;
+
+      case SortType.LENGTH:
+        visibleGoods.sort((firstGood, secondGood) => (
+          firstGood.length - secondGood.length
+        ));
+        break;
+
+      case SortType.NONE:
+      default:
+        break;
+    }
+
+    if (reversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  }
+
+  const handleAlphabet = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const handleLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const handleReverse = () => {
+    setReversed(!reversed);
+  };
+
+  const handleReset = () => {
+    setReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const sortedGoods = getReorderedGoods(goodsFromServer);
+  const isSorted = reversed || sortType !== SortType.NONE;
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button is-info',
+            { 'is-light': sortType !== SortType.ALPHABET },
+          )}
+          onClick={() => handleAlphabet()}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames(
+            'button is-success',
+            { 'is-light': sortType !== SortType.LENGTH },
+          )}
+          onClick={() => handleLength()}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames(
+            'button is-warning',
+            { 'is-light': !reversed },
+          )}
+          onClick={() => handleReverse()}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {isSorted && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => handleReset()}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {sortedGoods.map(good => (
+          <li key={good} data-cy="Good" className="good">
+            {good}
+          </li>
+        ))}
       </ul>
```

Comment:
maybe we need to create a separate component for this list. Just my opinion=))

Reply:
@kuzmi491 Good point.
@olecsiy-gavrilyuk Fix it too.